docs/general/ad-filtering/create-own-filters.md

```diff
@@ -1240,6 +1262,8 @@ Here's how it works:
 
 Rules with `$cookie` modifier are not supported by AdGuard Content Blocker, AdGuard for iOS, and AdGuard for Safari.
 
+In AdGuard Browser Extension MV3, the work of `$cookie` rules with parameters is done on the content script side. But since we can no longer hang a blocking listener on events, all of these actions happen asynchronously, and therefore not all changes to cookies will have time to be applied before they are sent to the server.
```

Comment:
The wording is not ideal.

@105th what did you try to tell us here?

Reply:
@ameshkov we are trying to delete cookies from 2 sides: from content-script (for which we have access) and from onBeforeSendHeaders listener. but since `onBeforeSendHeaders` and other listeners are not blocking anymore we have some race condition here: sometimes we managed to delete them before sending, sometimes not.

it can be checked with this test - https://testcases.agrd.dev/Filters/cookie-rules/test-cookie-rules.
first test here (blocks request cookie) sometimes fails, sometimes not.